state/protocol/inmem/epoch.go

```diff
@@ -235,23 +243,24 @@ func (e *heightBoundedEpoch) FinalHeight() (uint64, error) {
 	return 0, protocol.ErrUnknownEpochBoundary
 }
 
-// NewSetupEpoch returns a memory-backed epoch implementation based on an
-// EpochSetup event. Epoch information available after the setup phase will
-// not be accessible in the resulting epoch instance.
+// NewSetupEpoch returns a memory-backed epoch implementation based on an EpochSetup event.
+// Epoch information available after the setup phase will not be accessible in the resulting epoch instance.
 // No errors are expected during normal operations.
-func NewSetupEpoch(setupEvent *flow.EpochSetup) protocol.Epoch {
+func NewSetupEpoch(setupEvent *flow.EpochSetup, extensions []flow.EpochExtension) protocol.Epoch {
```

Comment:
In my opinion, this method should be deprecated for the scope of this Pr, including a TODO to remove it entirely

```suggestion
//
// DEPRECATED and TO BE REMOVED, because setupEpoch simply cannot be a valid representation of protocol.Epoch
// as it misses a bunch of the important information. We have updated our convention such that an epoch 
// essentially only starts to exist after the EpochCommit Service event has been finalized. The plan is to
// remove any exposure of uncommitted epochs from the `EpochQuery` API.
//
// No errors are expected during normal operations.
func NewSetupEpoch(setupEvent *flow.EpochSetup) protocol.Epoch {
```

Reply:
See https://github.com/onflow/flow-go/pull/6154#discussion_r1666853621